src/python/dxpy/sugar/objects.py

```diff
@@ -0,0 +1,439 @@
+                "Folder %s not found in project %s", folder, project.get_id(),
+                exc_info=True
+            )
+            raise dxpy.DXSearchError(
+                "Folder {} does not exist in project {}".format(
+                    folder, project.get_id()
+                )
+            )
+        elif create:
+            LOG.info("Creating new folder %s in project %s", folder, project.get_id())
+            project.new_folder(folder, parents=True)
+            return []
+    elif exists is False:
+        raise dxpy.DXSearchError(
+            "Folder {} exists in project {} but was expected not to exist".format(
+                folder, project.get_id()
+            )
+        )
+    else:
+        return ls
+
+
+def get_data_object(
+    data_obj_desc, project=None, classname="file", exists=True, **kwargs
+):
+    """Gets a data object by name or ID.
+
+    Args:
+        data_obj_desc: ID or name of the file, or an instance of DXDataObject.
+        project: The project in which to get the file. Defaults to the current project.
+        classname: Classname of the data object; defaults to "file" but can also be
+            "record" or "database".
+        exists: Assert whether the file exists (True) or does not exist (False). If
+            None, existance is not checked.
+        **kwargs: Additional keyword arguments to use when searching for file.
+
+    Returns:
+        A DXFile object.
+
+    Raises:
+        dxpy.DXSearchError if the data object does exist and `exists is False` or if
+        the data object does not exist and `exists is True`.
+    """
+    if data_obj_desc is None:
+        raise ValueError("'data_obj_desc' cannot be None")
+
+    project = get_project(project)
+    data_obj = None
+    search = True
+
+    if isinstance(data_obj_desc, dxpy.DXFile):
+        data_obj = data_obj_desc
+        search = False
+    elif resolver.is_data_obj_id(data_obj_desc):
+        data_obj = dxpy.get_handler(data_obj_desc, project)
+
+    if data_obj:
+        try:
+            data_obj.describe()
+        except dxpy.DXError:
+            data_obj = None
+
+    if data_obj is None and search:
+        folder, name, recurse = _parse_object_name(data_obj_desc)
+        kwargs["return_handler"] = True
+        if "folder" not in kwargs:
+            kwargs["folder"] = folder
+        if "recurse" not in kwargs:
+            kwargs["recurse"] = recurse
+        data_obj = dxpy.find_one_data_object(
+            zero_ok=True,
+            more_ok=False,
+            name=name,
+            classname=classname,
+            **kwargs
+        )
+
+    if not data_obj and exists is True:
+        raise dxpy.DXSearchError("File {} does not to exist", data_obj_desc)
+    if data_obj and exists is False:
+        raise dxpy.DXSearchError(
+            "File {} exists but was expected not to exist", data_obj_desc
+        )
+
+    return data_obj
+
+
+def get_workflow(workflow_desc, project=None, **kwargs):
```

Comment:
Good point. I updated get_workflow to call get_data_object when searching for a project-based workflow, and get_app_or_applet to call get_data_object when searching for an applet. I also added some text to the docstrings explaining the use case for these methods. I also removed the kwargs from get_workflow and get_app_or_applet since the allowed parameters differ depending on whether it is project-based or global.